gui/run_manager.py

```diff
@@ -45,13 +50,84 @@ def initialized(self):
         """
         return True if (self.validated_runs is not None and isinstance(self.validated_runs, dict)) else False
 
+    def set_config(self, filepath, type):
+        """
+        Sets the configuration filepath for run_list and for SPECjbb jar files.
+        Caller of `load_config()` (when Type=="RunList") to update configurations stored in memory.
+        :param filepath:
+        :param type:
+        :return:
+        """
+        if (filepath or type) is None:
+            return None
+        extension = pathlib.Path(filepath).suffix
+        if "json" in extension and type == "RunList":
+            if filepath != self.RUN_CONFIG:
+                if Path(filepath).is_file():  # todo: test
```

Comment:
We should add tests for these using `unittest`.

Reply:
Yeah, RunManager should be tested thoroughly since GUI is so dependent on it working correctly

Reply:
@austin-millan https://github.com/PDXCapstoneF/SPECtate/pull/151 opened for that purpose. Feel free to add more to the PR if there's stuff I missed.